### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by annabasenyuk**

### CODE REVIEW

**Review by mpohorenyi — APPROVED** [review]
Great Job!

File: src/makeCalculator.js [hunk 1]
```diff
@@ -37,7 +37,47 @@
...
+    },
+
+    reset() {
+      obj.result = 0;
+
+      return obj;
+    },
+
+    operate(calculator, num) {
+      calculator(num);
```
Comment (vasylmiskiv) [inline src/makeCalculator.js:74]: You should bind the calculator callback to the current context via `call` method

File: src/makeCalculator.js [hunk 2]
```diff
@@ -37,7 +37,47 @@
...
  */
 function makeCalculator() {
-  // write code here
+  const obj = {
+    result: 0,
+
+    add(num) {
+      obj.result += num;
+
+      return obj.result;
```
Comment (vasylmiskiv) [inline src/makeCalculator.js:46]: You should return just `this `  and to the same for the following operations

File: src/makeCalculator.js [hunk 3]
```diff
@@ -37,7 +37,47 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const obj = {
+    result: 0,
+
+    add(num) {
+      obj.result += num;
```
Comment (vasylmiskiv) [inline src/makeCalculator.js:44]: Use `this` instead of `obj`

**Review by vasylmiskiv — CHANGES_REQUESTED** [review]

File: src/makeCalculator.js (hunk 3 above)
Comment (annabasenyuk) [reply src/makeCalculator.js]: Hi) thanks for the comments, I hope I understood everything correctly and corrected it)

**Pull request closed without merging**